crates/storage/provider/src/traits/storage.rs

```diff
@@ -0,0 +1,33 @@
+use std::{
+    collections::{BTreeMap, BTreeSet},
+    ops::RangeInclusive,
+};
+
+use auto_impl::auto_impl;
+use reth_interfaces::Result;
+use reth_primitives::{Address, BlockNumber, StorageEntry, H256};
+
+/// Storage reader
+#[auto_impl(&, Arc, Box)]
+pub trait StorageReader: Send + Sync {
+    /// Get plainstate storages from an Address.
+    fn basic_storages(
```

Comment:
I find this naming here a bit confusing.

what does `basic` mean here?

`basic` returns `plainstate` is not very intuitive to me